src/index.ts

```diff
@@ -0,0 +1,263 @@
+import { RLPInput, RLPDecoded } from "./types";
+import BN from 'bn.js';
+
+/**
+ * RLP Encoding based on: https://github.com/ethereum/wiki/wiki/%5BEnglish%5D-RLP
+ * This function takes in a data, convert it to buffer if not, and a length for recursion
+ * @param input - will be converted to buffer
+ * @returns returns buffer of encoded data
+ **/
+export function encode(input: RLPInput): Buffer {
+  if (input instanceof Array) {
+    const output: Buffer[] = [];
+    for (let i = 0; i < input.length; i++) {
+      output.push(encode(input[i]));
+    }
+    const buf = Buffer.concat(output);
+    return Buffer.concat([encodeLength(buf.length, 192), buf]);
+  } else {
+    const inputBuf = toBuffer(input);
+    return inputBuf.length === 1 && inputBuf[0] < 128
+      ? inputBuf
+      : Buffer.concat([encodeLength(inputBuf.length, 128), inputBuf]);
+  }
+}
+
+/**
+ * Parse integers. Check if there is no leading zeros
+ * @param v The value to parse
+ * @param base The base to parse the integer into
+ */
+function safeParseInt(v: string, base: number): number {
+  if (v.slice(0, 2) === "00") {
+    throw new Error("invalid RLP: extra zeros");
+  }
+
+  return parseInt(v, base);
+}
+
+function encodeLength(len: number, offset: number): Buffer {
+  if (len < 56) {
+    return Buffer.from([len + offset]);
+  } else {
+    const hexLength = intToHex(len);
+    const lLength = hexLength.length / 2;
+    const firstByte = intToHex(offset + 55 + lLength);
+    return Buffer.from(firstByte + hexLength, "hex");
+  }
+}
+
+/**
+ * RLP Decoding based on: {@link https://github.com/ethereum/wiki/wiki/%5BEnglish%5D-RLP|RLP}
+ * @param input - will be converted to buffer
+ * @param stream - Is the input a stream (false by default)
+ * @returns - returns decode Array of Buffers containg the original message
+ **/
+export function decode(input: RLPInput, stream: boolean = false): Buffer[] | Buffer | RLPDecoded {
+  if (!input || (<any>input).length === 0) {
+    return Buffer.from([]);
+  }
+
+  const inputBuffer = toBuffer(input);
+  const decoded = _decode(inputBuffer);
+
+  if (stream) {
+    return decoded;
+  }
+  if (decoded.remainder.length !== 0) {
+    throw new Error("invalid remainder");
+  }
+
+  return decoded.data;
+}
+
+/**
+ * Get the length of the RLP input
+ * @param input
+ * @returns The length of the input or an empty Buffer if no input
+ */

```

Comment:
Is this intended to actually remove the types from the comments here? Does TypeScript do some auto-magic here?

Reply:
yes. tsdoc knows the type from the code. It would throw a warning if I didn't remove it.

Reply:
Ah ok, thanks for clarification, just also had a look in the TSDoc playground. I find it a bit strange that types then seems to be not part of the generated documentation and not completely getting the concept behind it. But if this is the official documentation format, so be it.